### PR TITLE
Postgresql flexible

### PR DESCRIPTION
**Pull request by schwichti**
- Replace Azure Database for PostgreSQL single server by flexible server https://github.com/dspace-group/simphera-reference-architecture-azure/issues/45
- Make Kubernetes Tier configurable https://github.com/dspace-group/simphera-reference-architecture-azure/issues/88
- Replace antonbabenko/pre-commit-terraform with docker_image hooks as it did not work properly on Windows https://github.com/dspace-group/simphera-reference-architecture-azure/issues/60
- Enable geo redundant backups for PostgreSQL https://github.com/dspace-group/simphera-reference-architecture-azure/issues/36
- Enable restore retention of storage accounts https://github.com/dspace-group/simphera-reference-architecture-azure/issues/25
- Enable Azure AD authentication in Azure Database for PostgreSQL by default (required when using Azure Managed Identity)

### CODE REVIEW

File: hooks/validate_tfvars.sh [hunk 1]
```diff
@@ -3,11 +3,14 @@
 #SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
 # run terraform apply and redirect stderr to stdout
 cd /src
-OUTPUT=`terraform apply --var-file="terraform.tfvars.example" -input=false 2>&1 >/dev/null`
+terraform init -lock=false
+OUTPUT=`terraform apply -lock=false --var-file="terraform.tfvars.example" -input=false 2>&1 >/dev/null`
 echo $OUTPUT
 # ignore az login
 if [[ $OUTPUT == *"az login"* ]]; then
+  echo "here"
```
Comment (mariogluhakovic) [inline hooks/validate_tfvars.sh:11]: Please remove debug lines.

File: hooks/validate_tfvars.sh [hunk 2]
```diff
@@ -3,11 +3,14 @@
...
   exit 0
 else
+  echo "here2"
```
Comment (mariogluhakovic) [inline hooks/validate_tfvars.sh:14]: Please remove debug lines.

File: terraform.tfvars.example [hunk 3]
```diff
@@ -69,7 +69,7 @@ logAnalyticsWorkspaceName = ""
 logAnalyticsWorkspaceResourceGroupName = ""
 
 # The version of the AKS cluster.
-kubernetesVersion = "1.24.9"
+kubernetesVersion = "1.24.10"
```
Comment (mariogluhakovic) [inline terraform.tfvars.example:72]: Did you maybe try to upgrade version to 1.26 and test these changes?

Comment (schwichti) [reply terraform.tfvars.example:72]: I suggest to create a separate PR.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -1,19 +1,32 @@
-# Rerun `pre-commit install` after every update of this file
 repos:
--   repo: https://github.com/antonbabenko/pre-commit-terraform
-    rev: v1.77.0
+-   repo: local
+    hooks:
+    -   id: terraform_format
+        name: terraform_format
+        entry: --entrypoint /src/hooks/terraform_format.sh hashicorp/terraform:1.5.2
```
Comment (mariogluhakovic) [inline .pre-commit-config.yaml:6]: Can we use inline command call or the bash file with one command call is needed?

File: hooks/validate_tfvars.sh (hunk 1 above)
Comment (schwichti) [reply hooks/validate_tfvars.sh]: done :see_no_evil:

File: terraform.tfvars.example (hunk 3 above)
Comment (schwichti) [reply terraform.tfvars.example:72]: done :see_no_evil:

File: .pre-commit-config.yaml (hunk 4 above)
Comment (schwichti) [reply .pre-commit-config.yaml:6]: Sadly, unline command did not work for me.

Comment (schwichti) [reply .pre-commit-config.yaml:6]: I believe that something goes wrong when pre-commit passes docker arguments. I was not able to find it out in time.

**Review by mariogluhakovic — APPROVED** [review]

**Pull request merged**